examples/custom_shaders_uniforms/GlutWindow.cc

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#if __APPLE__
+  #include <OpenGL/gl.h>
+  #include <OpenGL/OpenGL.h>
+  #include <GLUT/glut.h>
+#else
+  #include <GL/glew.h>
+  #include <GL/gl.h>
+  #include <GL/glut.h>
+#endif
+
+#if !defined(__APPLE__) && !defined(_WIN32)
+  #include <GL/glx.h>
+#endif
+
+#include <mutex>
+
+#include <ignition/common/Console.hh>
+#include <ignition/rendering.hh>
+#include <ignition/rendering/Camera.hh>
+#include <ignition/rendering/Image.hh>
+#include <ignition/rendering/Scene.hh>
+
+#include "GlutWindow.hh"
+
+#define KEY_ESC 27
+#define KEY_TAB  9
+
+//////////////////////////////////////////////////
+unsigned int imgw = 0;
+unsigned int imgh = 0;
+
+std::vector<ir::CameraPtr> g_cameras;
+ir::CameraPtr g_camera;
+ir::CameraPtr g_currCamera;
+unsigned int g_cameraIndex = 0;
+ir::ImagePtr g_image;
+
+bool g_initContext = false;
+
+#if __APPLE__
+  CGLContextObj g_context;
+  CGLContextObj g_glutContext;
+#elif _WIN32
+#else
+  GLXContext g_context;
+  Display *g_display;
+  GLXDrawable g_drawable;
+  GLXContext g_glutContext;
+  Display *g_glutDisplay;
+  GLXDrawable g_glutDrawable;
+#endif
+
+double g_offset = 0.0;
+
+int g_seed[1] = {0};
+float g_resolution[2] = {400, 200};
+float g_color[3] = {1.0, 1.0, 1.0};
+float g_adjustments[16] = {
+  0, 0, 0, 0.0005,
+  0, 0, 0, 0,
+  0, 0, 0, 0,
+  0, 0, 0, 0
+};
+
+//////////////////////////////////////////////////
+//! [update camera]
+void updateCameras()
+{
+  double angle = g_offset / 2 * M_PI;
+  double x = sin(angle) * 3.0 + 3.0;
+  double y = cos(angle) * 3.0;
+  for (ir::CameraPtr camera : g_cameras)
+  {
+    camera->SetLocalPosition(x, y, 0.0);
+  }
+
+  g_offset += 0.0005;
+}
+//! [update camera]
+
+//////////////////////////////////////////////////
+//! [update uniforms]
+void updateUniforms()
+{
+  ir::NodePtr node = g_camera->Parent();
+  ir::VisualPtr sphere = std::dynamic_pointer_cast<ir::Visual>(node->ChildByName("box"));
+  ir::MaterialPtr shader = sphere->Material();
+  ir::ShaderParamsPtr shaderParams = shader->FragmentShaderParams();
+  
+  (*shaderParams)["u_seed"].InitializeWordBuffer(1);
```

Comment:
this reset the buffer every iteration. Maybe we can have an `initUniforms` function that calls the initialization only once?